src/NuGet.Core/NuGet.Protocol/Resources/OwnerDetailsUriTemplateResourceV3.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Protocol.Resources
+{
+    /// <summary>Owner Details Uri Template for NuGet V3 HTTP feeds.</summary>
+    /// <remarks>Not intended to be created directly. Use <see cref="SourceRepository.GetResourceAsync{T}(CancellationToken)"/>
+    /// with <see cref="OwnerDetailsUriTemplateResourceV3"/> for T, and typecast to this class.
+    public class OwnerDetailsUriTemplateResourceV3 : INuGetResource
+    {
+        private readonly string _template;
+
+        private OwnerDetailsUriTemplateResourceV3(string template)
+        {
+            _template = template ?? throw new ArgumentNullException(nameof(template));
+        }
+
+        /// <summary>
+        /// Creates the specified Owner Details Uri template provided by the server if it exists and is valid.
+        /// </summary>
+        /// <param name="uriTemplate">The Absolute Uri template provided by the server.</param>
+        /// <returns>A valid Owner Details Uri template, or null.</returns>
+        public static OwnerDetailsUriTemplateResourceV3? CreateOrNull(Uri uriTemplate)
+        {
+            if (uriTemplate is null)
+            {
+                throw new ArgumentNullException(nameof(uriTemplate));
+            }
+
+            if (uriTemplate.OriginalString.Length == 0
+                || !uriTemplate.IsAbsoluteUri
+                || !uriTemplate.Scheme.Equals("https", StringComparison.OrdinalIgnoreCase))
+            {
+                return null;
+            }
+
+            return new OwnerDetailsUriTemplateResourceV3(uriTemplate.OriginalString);
+        }
+
+        /// <summary>
+        /// Gets a URL for viewing package Owner URL outside of Visual Studio. The URL will not be verified to exist.
+        /// </summary>
+        /// <param name="owner">The owner username.</param>
+        /// <returns>The first URL from the resource, with the URI template applied.</returns>
+        public Uri GetUri(string owner)
+        {
+            var uriString = _template
+#if NETCOREAPP
+               .Replace("{owner}", owner, StringComparison.OrdinalIgnoreCase);
```

Comment:
The `string.Replace()` method in this function replaces the `{owner}` substring only if it exists; otherwise, it does nothing and returns the same string. Given that nuget.org includes the `{owner}` substring in the template URL and we are planning to make the placeholder mandatory as per [this discussion](https://github.com/NuGet/docs.microsoft.com-nuget/pull/3282/files#r1576870144), we seem to lack necessary client-side validations.

I also suggest adding an additional check to ensure that the actual URL derived from replacing the `{owner}` placeholder remains within the confines of the specified template URL, which could help protect against path traversal issues. Here's an illustrative example demonstrating this:

```cs
Uri templateUri = new Uri("https://nuget.test/profiles/{owner}?_src=template");
Uri actualUri = new Uri("https://nuget.test/profiles/kp?_src=template");
Console.WriteLine("IsBaseOf: " + templateUri.IsBaseOf(actualUri)); // returns true

Uri templateUri = new Uri("https://nuget.test/profiles/{owner}?_src=template");
Uri actualUri = new Uri("https://nuget.test/profiles/../kp?_src=template");
Console.WriteLine("IsBaseOf: " + templateUri.IsBaseOf(actualUri)); // returns false because "../kp" string is specified as owner which changes the URL to "https://nuget.test/kp?_src=template" after resolving.
```

This additional check using `IsBaseOf` could be crucial for maintaining the integrity and security of URL generation. I found out about the `IsBaseOf` method in [Security considerations](https://learn.microsoft.com/dotnet/fundamentals/runtime-libraries/system-uri#security-considerations) from the docs.

Reply:
The outcome of the discussion is in [this comment](https://github.com/NuGet/docs.microsoft.com-nuget/pull/3282/files#r1576889184) is to document that it's required, but that Client will not require it. Therefore, it's acceptable that a Uri template not contain `{owner}`.

>  2) we say it's required in the docs because we think it should really be there, but we make no effort to enforce it in the client.

Would the `actualUri` in your example fail to be an absolute URI in that case? I'll have to test this and follow-up if necessary. Note that Package Details Uri has used this pattern for the template since it was created.